### ISSUE TITLE
This extension is breaking my VSCode

### ISSUE THREAD

**Issue by kriscoleman** [formulahendry/vscode-auto-rename-tag#48]
Hello, 

I had some strange happenings going on in my VS Code: 

- source control tab would stop tracking changes... modified files wouldn't update the changed files list on this tab. clicking on a file to showdiff would also not do anything. 
- vs vim would stop working 
- intellisense and autocomplete would break 

I looked everywhere for errors pertaining to this issue but couldn't find any anywhere. 

I eventually started disabling all my extensions one-by-one. When I disabled this one, everything started working find again. 

I wanted to report the issue here to make you aware. I'm sorry I don't have more information.

**Comment by allevaton** [comment]
I am in the same boat @kriscoleman. Recently re-enabled all my extensions one by one to find the problem and ended up here.

I'm getting absurd, disruptive, long pauses in the entire VS Code process when changing selections, typing, even opening files. It stopped when I disabled this extension.

**Comment by DifficultNick** [comment]
Here is an example for you when everything breaks.
Self-closed tag
```XML
<Tag Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1" Attr="1"/>
```
It seems that the more attributes it has, the slower extension will work.

**Comment by jhixson** [comment]
+1 for this issue. I thought it was the Vim plugin that was slowing down but this seems to be the culprit. Similar to @DifficultNick I am only seeing excessive CPU usage on very long lines with lots of attributes (e.g. custom React JSX elements).

**Comment by evenfrost** [comment]
Same here for `.vue` files. VS Code is starting to use 100% CPU on all cores when working on long tags with Auto Rename Tag enabled. And it crashes VSCodeVim at first for me. Here's my example tag:
```html
 <path fill="#fff" d="M24.016 10.703a7.103 7.103 0 0 1-2.045.56 3.57 3.57 0 0 0 1.566-1.969 7.126 7.126 0 0 1-2.26.864 3.56 3.56 0 0 0-6.066 3.247 10.105 10.105 0 0 1-7.339-3.72 3.56 3.56 0 0 0 1.102 4.753 3.535 3.535 0 0 1-1.613-.446v.046a3.562 3.562 0 0 0 2.856 3.49 3.547 3.547 0 0 1-1.608.062 3.564 3.564 0 0 0 3.326 2.472 7.143 7.143 0 0 1-5.27 1.474 10.07 10.07 0 0 0 5.456 1.6c6.548 0 10.129-5.425 10.129-10.13 0-.154-.004-.308-.01-.46a7.22 7.22 0 0 0 1.776-1.843z" />
```

**Comment by explorador** [comment]
+1. Happens when a hit "tab" before an svg tag.

**Comment by mojosef** [comment]
+1. Narrowed down to this plugin ( which is great ). Seems to be when the file has tags containing many characters. Not limited to any file type for me.

VS Code Version: 1.22.2 (1.22.2)
Plugin Version: 0.0.15 (2017-11-04)
MacPro: 10.13.3

**Comment by ericloong90** [comment]
I'm having similar issue as @kriscoleman after disabling one extension at a time. When I was using vs-vim, this extension would bring Code Helper's CPU usage to 100%, and reloading window would just create another instance of Code Helper, which then would just shoot up to 100% again. Once I had 6 Code Helper instances all running at full CPU usage. Disabling this extension until further notice.

**Comment by hassanbinsafdar** [comment]
same goes for me solve this issue its a great extension but please this not bearable so i disabled and not using anymore

**Comment by justablob** [comment]
This needs to be resolved. I thought i was the only one getting this.

**Comment by hassanbinsafdar** [comment]
@justablob same here i think they are not interested to solve this anymore so try to search alternate :(

**Comment by hassanbinsafdar** [comment]
@justablob 
This is other extension its not automatic but it can give some better things.
https://marketplace.visualstudio.com/items?itemName=krizzdewizz.tag-rename

**Comment by jonmwood** [comment]
Yes, this happens when in my JSX files especially when appending styles:

`<SomeDiv style={{ position: 'absolute', top: '0.6rem', left: '0.6rem', zIndex: '5', color: '#bbb' }} />`
